### DISCUSSION TITLE
How to pass function into component? It is possible?

### DISCUSSION THREAD

**Discussion by myrpg-fun** [DioxusLabs/dioxus#3107] [Q&A]
How to pass function into component? It is possible?

```
#[component]
fn UserSelect(
   onselect: impl FnMut(...)
) -> Element {
```
?

**Comment by ryo33** [comment]
For that case, it seems you can use [EventHandler](https://docs.rs/dioxus/0.6.0-alpha.3/dioxus/prelude/type.EventHandler.html) as a props type as https://dioxuslabs.com/learn/0.5/reference/event_handlers.

Also, you could use [Callback](https://docs.rs/dioxus/0.6.0-alpha.3/dioxus/prelude/struct.Callback.html).

**Comment by wiseaidev** [comment]
just provide a regular rusty function type definition like so:

```rust
#[component]
fn UserSelect(
   onselect: fn(&str) -> bool,
) -> Element {
```

Here the `onselect` closure you are trying to pass accepts an `&str` param and returns a boolean. Learned this while building this [custom `input` component](https://github.com/opensass/aibook/blob/main/src/components/dashboard/fields/input.rs).
